src/main.ts

```diff
@@ -42,10 +43,15 @@ fetchResultSet$.subscribe((result) => {
   if (!result || !result.result){
     console.error("Results have not been returned.");
   }
-  if (location === "custom") {
+
+  if (localStorage.getItem("location") === "custom") {
     hideForecastLoader("custom");
   }
-  displayWeatherForecast(location, result.result.dataseries);
+
+  if (localStorage.getItem("location")) {
```

Comment:
What happens if you cannot get the location?

Reply:
I do set the location to Pretoria by default to ensure it is never empty, but I have added a guard clause just in case.